### PR TITLE
Simplify and split interface for executor services [PLEN-96]

### PR DESCRIPTION
**Pull request by nicu-da**
This is done to simplify integration into canton.

### CODE REVIEW

File: libs-scala/executors/src/main/scala/com/daml/executors/NamedExecutor.scala [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.executors.executors
+
+trait NamedExecutor {
+
+  /** Unique name for the executor service that can be used as identifier in logs and metrics
+    */
+  def name: String
```
Comment (simonmaxen-da) [inline libs-scala/executors/src/main/scala/com/daml/executors/NamedExecutor.scala:10]: How about changing the type of this to `MetricName` ?

**Review by simonmaxen-da — APPROVED** [review]

File: libs-scala/executors/src/main/scala/com/daml/executors/NamedExecutor.scala (hunk 1 above)
Comment (simonmaxen-da) [reply libs-scala/executors/src/main/scala/com/daml/executors/NamedExecutor.scala:10]: The intention is that the self service error codes associated with rate limiting have a reference to the _metric name_.  A metric label is next to useless unless the name that is known. Additionally the same label could be applied to many metrics.

Comment (simonmaxen-da) [reply libs-scala/executors/src/main/scala/com/daml/executors/NamedExecutor.scala:10]: Maybe both are needed ?

Comment (simonmaxen-da) [reply libs-scala/executors/src/main/scala/com/daml/executors/NamedExecutor.scala:10]: The idea is that from the self service error code an operator can go into Grafana and display the metric with that name and see where the resource being limited is breaching the limit over time.

Comment (nicu-da) [reply libs-scala/executors/src/main/scala/com/daml/executors/NamedExecutor.scala:10]: Yes, so the name is already referenced, but it's not this one, it's a constant one for metrics.
Then for the name the label has to be used to know for which pool is the metrics. Only the label is dynamic and per pool.

**Pull request merged**